modules/atmosphere/rendering/atmospheredeferredcaster.cpp

```diff
@@ -0,0 +1,1595 @@
+        _inScatteringSupTermsProgramObject = nullptr;
+    }
+
+    if (_deltaEProgramObject) {
+        _deltaEProgramObject = nullptr;
+    }
+
+    if (_deltaSProgramObject) {
+        _deltaSProgramObject = nullptr;
+    }
+
+    if (_deltaSSupTermsProgramObject) {
+        _deltaSSupTermsProgramObject = nullptr;
+    }
+
+    if (_deltaJProgramObject) {
+        _deltaJProgramObject = nullptr;
+    }
+
+    glDeleteTextures(1, &_transmittanceTableTexture);
+    glDeleteTextures(1, &_irradianceTableTexture);
+    glDeleteTextures(1, &_inScatteringTableTexture);
+    glDeleteTextures(1, &_deltaETableTexture);
+    glDeleteTextures(1, &_deltaSRayleighTableTexture);
+    glDeleteTextures(1, &_deltaSMieTableTexture);
+    glDeleteTextures(1, &_deltaJTableTexture);
+    glDeleteTextures(1, &_atmosphereTexture);
+
+}
+
+void AtmosphereDeferredcaster::preRaycast(const RenderData & renderData, const DeferredcastData& deferredData,
+                                          ghoul::opengl::ProgramObject& program) 
+{    
+    // Atmosphere Frustum Culling
+    glm::dvec3 tPlanetPosWorld = glm::dvec3(_modelTransform * glm::dvec4(0.0, 0.0, 0.0, 1.0));
+
+    if (glm::distance(tPlanetPosWorld, renderData.camera.positionVec3()) > DISTANCE_CULLING) {
+        program.setUniform("cullAtmosphere", 1);
+    }
+    else {
+        glm::dmat4 MV = glm::dmat4(renderData.camera.sgctInternal.projectionMatrix()) * renderData.camera.combinedViewMatrix();
+        if (!isAtmosphereInFrustum(glm::value_ptr(MV), tPlanetPosWorld, (_atmosphereRadius + 2.0)*1000.0)) {
+            program.setUniform("cullAtmosphere", 1);
+        }
+        else {
+            program.setUniform("cullAtmosphere", 0);
+            program.setUniform("Rg", _atmospherePlanetRadius);
+            program.setUniform("Rt", _atmosphereRadius);
+            program.setUniform("AverageGroundReflectance", _planetAverageGroundReflectance);
+            program.setUniform("HR", _rayleighHeightScale);
+            program.setUniform("betaRayleigh", _rayleighScatteringCoeff);
+            program.setUniform("HM", _mieHeightScale);
+            program.setUniform("betaMieScattering", _mieScatteringCoeff);
+            program.setUniform("betaMieExtinction", _mieExtinctionCoeff);
+            program.setUniform("mieG", _miePhaseConstant);
+            program.setUniform("sunRadiance", _sunRadianceIntensity);
+            program.setUniform("ozoneLayerEnabled", (bool)_ozoneEnabled);
+            program.setUniform("HO", _ozoneHeightScale);
+            program.setUniform("betaOzoneExtinction", _ozoneExtinctionCoeff);
+
+            program.setUniform("exposure", _exposureConstant);
+            program.setUniform("backgroundExposure", _exposureBackgroundConstant);
+            program.setUniform("gamma", _gammaConstant);
+            program.setUniform("RenderableClass", static_cast<int>(_renderableClass));
+
+            program.setUniform("TRANSMITTANCE_W", (int)_transmittance_table_width);
```

Comment:
Actually they were defined as unsigned int and I was being picky. Changed to int and removed the casting.